aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsProxyHttpServletRequest.java

```diff
@@ -471,6 +480,11 @@ public RequestDispatcher getRequestDispatcher(String s) {
 
     @Override
     public int getRemotePort() {
+        if (Objects.nonNull(request.getRequestContext().getElb())) {
+            String hostHeader = request.getHeaders().get(PORT_HEADER_NAME);
```

Comment:
Variable naming is misleading as it is no longer the host header.

Reply:
Updated. Thanks.

Reply:
I think you're right. We need to update getRemoteAddr() to take into account the x-forwarded-for header. For getServerPort(), I'm a bit confussed. Looking at the javadoc for both getServerPort() and getRemotePort(), I'm wondering if `Returns the Internet Protocol (IP) source port of the client or last proxy that sent the request.` and `Returns the port number to which the request was sent` don't mean the same thing. Is ALB considered as a client in this case?

Reply:
To make it even more interesting there is also `getLocalPort()`.

ALB is the last proxy so that should be fine.

For comparison Tomcat sets `serverPort` based on scheme https://github.com/apache/tomcat/blob/main/java/org/apache/catalina/connector/CoyoteAdapter.java#L579.